backend/telemetry_shard/src/json_message/hash.rs

```diff
@@ -0,0 +1,214 @@
+use serde::de::{self, Deserialize, Deserializer, SeqAccess, Unexpected, Visitor};
+use serde::ser::{Serialize, Serializer};
+use std::fmt::{self, Debug, Display};
+use std::str::FromStr;
+
+const HASH_BYTES: usize = 32;
+
+/// Newtype wrapper for 32-byte hash values, implementing readable `Debug` and `serde::Deserialize`.
+/// This can deserialize from a JSON string or array.
```

Comment:
(this was copied over from the previous version; I think the point of this is that it can read hashes from hex strings as well as from arrays of bytes, and present hashes in such a way when debugging)

Reply:
Cool, yeah, let's add that!

Reply:
Done :)